collect_app/src/main/java/org/odk/collect/android/mainmenu/MainMenuActivity.kt

```diff
@@ -31,48 +30,69 @@ import org.odk.collect.android.utilities.ApplicationConstants
 import org.odk.collect.android.utilities.PlayServicesChecker
 import org.odk.collect.android.utilities.ThemeUtils
 import org.odk.collect.androidshared.ui.DialogFragmentUtils.showIfNotShowing
+import org.odk.collect.androidshared.ui.FragmentFactoryBuilder
 import org.odk.collect.androidshared.ui.multiclicksafe.MultiClickGuard.allowClick
 import org.odk.collect.crashhandler.CrashHandler
+import org.odk.collect.permissions.PermissionsChecker
+import org.odk.collect.permissions.PermissionsProvider
 import org.odk.collect.projects.Project.Saved
 import org.odk.collect.settings.SettingsProvider
 import org.odk.collect.settings.keys.ProjectKeys
 import org.odk.collect.strings.localization.LocalizedActivity
 import javax.inject.Inject
 
 class MainMenuActivity : LocalizedActivity() {
-    @Inject
-    lateinit var viewModelFactory: MainMenuViewModel.Factory
 
     @Inject
-    lateinit var currentProjectViewModelFactory: CurrentProjectViewModel.Factory
+    lateinit var viewModelFactory: MainMenuViewModelFactory
 
     @Inject
     lateinit var settingsProvider: SettingsProvider
 
+    @Inject
+    lateinit var permissionsProvider: PermissionsProvider
+
+    @Inject
+    lateinit var permissionChecker: PermissionsChecker
+
     private lateinit var binding: MainMenuBinding
     private lateinit var mainMenuViewModel: MainMenuViewModel
     private lateinit var currentProjectViewModel: CurrentProjectViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         initSplashScreen()
-        super.onCreate(savedInstanceState)
-        binding = MainMenuBinding.inflate(layoutInflater)
 
         CrashHandler.getInstance(this)?.also {
             if (it.hasCrashed(this)) {
+                super.onCreate(null)
```

Comment:
`onCreate()` has to call `super.onCreate()` before returning.

Reply:
Could you elaborate on this change? Why `super.onCreate(savedInstanceState)` just after `initSplashScreen()` wasn't enough?

Reply:
It's because of fragments being restored. We need to setup the `FragmentFactory` before `super.onCreate` so that Fragments can be reconstructed on rotation etc. We then also need `inject` before `super.onCreate` as injected fields will get used to create the fragments. Because we also want our `CrashHandler` stuff before `inject` to deal with cases where we can't construct the `AppDependencyComponent (or dependency within it), we now need `CrashHandler.getInstance` above `super.onCreate`.